test/test_sensors.py

```diff
@@ -25,15 +25,17 @@ def _random_episode(env, config):
         0,
         np.cos(random_heading / 2),
     ]
-    env.episodes = [
-        NavigationEpisode(
-            episode_id="0",
-            scene_id=config.SIMULATOR.SCENE,
-            start_position=random_location,
-            start_rotation=random_rotation,
-            goals=[],
-        )
-    ]
+    env.episode_iterator = iter(
+        [
```

Comment:
If possible, you can skip creating the list here and just use a basic generator.

Reply:
@abhiskk  for now I think `iter([Episode])` is straightforward enough and requires minimal changes to test scripts. An alternative I can think of is maybe `(Episode for _ in range(1))`, which isn't much better IMO. I'll leave this unchanged if you agree.

I think for longer list of episodes, it's definitely worth it to create generator instead. And I can add a helper function to create such generator when I add tests to new features in next PR.

Reply:
this is fine, my mistake.